contrib/platform/test/com/sun/jna/platform/win32/Shell32Test.java

```diff
@@ -131,16 +135,89 @@ public void testResizeDesktopFromTop() throws InterruptedException {
 
     }
 
-    public void testSHGetKnownFolderPath()
-    {
-        int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
-        PointerByReference outPath = new PointerByReference();
-        HANDLE token = null;
-        GUID guid = KnownFolders.FOLDERID_Fonts;
-        HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
-
-        Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
-
-        assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
-    }
+	public void testSHGetKnownFolderPath() {
+		int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
+		PointerByReference outPath = new PointerByReference();
+		HANDLE token = null;
+		GUID guid = KnownFolders.FOLDERID_Fonts;
+		HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
+
+		Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
+
+		assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
+	}
+
+	public void testSHEmptyRecycleBin() {
+		File file = new File(System.getProperty("java.io.tmpdir") + System.nanoTime() + ".txt");
+		try {
+			// Create a file and immediately send it to the recycle bin.
+			try {
+				fillTempFile(file);
+				W32FileUtils.getInstance().moveToTrash(new File[] { file });
+			} catch (IOException e) {
+				throw new RuntimeException(e);
+			}
+
+			int result = Shell32.INSTANCE.SHEmptyRecycleBin(null, null,
+					Shell32.SHERB_NOCONFIRMATION | Shell32.SHERB_NOPROGRESSUI | Shell32.SHERB_NOSOUND);
+			// for reasons I can not find documented, the function returns the
+			// following:
+			// -2147418113 when the recycle bin has no items in it
+			// 0 when the recycle bin has items in it
+			assertTrue("Result should have been 0 when emptying Recycle Bin - there should have been a file in it.",
+					result == 0);
```

Comment:
Use _assertEquals_ in case the non-zero value can help diagnose why the test failed.